vela-templates/definitions/usage-examples/application-with-hpa.yaml

```diff
@@ -0,0 +1,45 @@
+apiVersion: core.oam.dev/v1beta1
+kind: Application
+metadata:
+  name: helloworld
+spec:
+  components:
+    - name: helloworld
+      type: webservice
+      properties:
+        cpu: "0.5"
+        exposeType: ClusterIP
+        image: oamdev/hello-world
+        memory: 1024Mi
+        ports:
+          - expose: true
+            port: 80
+            protocol: TCP
+      traits:
+        - type: scaler
+          properties:
+            replicas: 1
+        - type: hpa
+          properties:
+            targetAPIVersion: apps/v1
+            targetKind: Deployment
+            max: 10
+            min: 1
+            cpu:
+              type: Utilization
+              value: 80
+            mem:
+              type: AverageValue
+              value: 90
+            podCustomMetrics:
+              # here are custom metric names and values
+              - name: pod_net_received_rate
+                value: "77"
+              - name: pod_net_transmitted_rate
+                value: "88"
+              - name: pod_net_received_packets_rate
+                value: "95"
+              - name: pod_net_transmitted_packets_rate
+                value: "99"
+
```

Comment:
the example maybe not work, refer to https://kubevela.net/docs/tutorials/auto-scaler

Reply:
1. for cpu/mem metric, user need install metric-server.  Have updated my test procedure
```
intall velaUX and follow https://github.com/kubernetes-sigs/metrics-server/blob/master/README.md to install metric-server on my MacBook
```

2.for custom metrics, user should use their own metrics. Have add comment to notice user
```
              # here are custom metric names and values. Please replace them to be your metrics

```

Reply:
I mean:

> When you're using HPA, you must specify the [apply-once](https://kubevela.net/docs/end-user/policies/apply-once) policy to let the HPA control the replica field.

So the example here may cause confusion.

Reply:
fixed, add apply-once Trait